Some additional polishing of the splash implementation

alfonsosanchezbeato opened the pull request:
Some additional polishing of the splash implementation

alfonsosanchezbeato committed:
Move files in {bin/,sbin/,lib/} to usr/

Move these folders to ease integration in the base as as nowadays they
are just symbolic links.

alfonsosanchezbeato committed:
Take custom theme from ubuntu-data

alfonsosanchezbeato committed:
Set vendor path to the data partition

Set vendor path to the data partition, so we make sure we get a theme
from an encrypted partition.

alfonsosanchezbeato committed:
Change location for configuration file

Change location for configuration file to
/usr/share/plymouth/plymouthd.defaults, as that is expected to be the
distro default.

alfonsosanchezbeato committed:
Add plymouth-quit-wait.service so console-conf actually waits on it

alfonsosanchezbeato committed:
Add After=plymouth-start.service dep for plymouth-quit-wait

alfonsosanchezbeato committed:
Change name of snap to core-splash20

mvo5 commented on config/usr/share/plymouth/plymouthd.defaults:
```diff
@@ -0,0 +1,3 @@
+[Daemon]
+Theme=vendor
+ThemeDir=/run/mnt/data/plymouth/themes
```
Right now /run/mnt/data/ contains "system-data", "user-data" which are mapped to "/" and "/home" respectively. Is the thinking behind this that there should be a third hierarchy or could this be integrated into the system-data/ subtree? If it gets integrated some "writable-path" changes may be needed. But maybe I'm missing the bigger pictures, how will the data get into that dir in the first place? Is it via gadget.yaml?

alfonsosanchezbeato replied:
Yes, the plan is to be able to customize the theme by including the content from the gadget. I'm putting this in the data partition (encrypted) to avoid possible security issues as we discussed previously. Looking at this now, it is true that having a `plymouth` folder at the same level as `system-data` and `user-data` is a bit strange... but not sure if integrating it in `system-data` would make sense. Maybe a new folder like `gadget-data` would make sense?

mvo5 replied:
Thanks for elaborating this. I like the new design to put it there but agree that maybe "plymouth" is a bit too specific. I like `gadget-data` - maybe @niemeyer or @pedronis have an opinion about the name? In general we are very careful about (user visible) naming so I would love to get the input of one of them here.

alfonsosanchezbeato committed:
Make sure we also stop plymouth in recovery mode

alfonsosanchezbeato committed:
Support env vars to change fixed paths

Add support to using environent variables so we run plymouth from a
different sysroot. The environment variables are:

* PLYMOUTH_PLUGIN_PATH
* PLYMOUTH_THEME_PATH
* PLYMOUTH_POLICY_DIR

Which can be set to override the default directories with the same
name in plymouth's configure script.

akshaybabloo commented:
Is this something that will be merged soon?
